### ISSUE TITLE
`SimplifyDeep` doesn't deep simplify lists

### ISSUE THREAD

**Issue by KubaJastrz** [sindresorhus/type-fest#884]
```ts
interface Interface_List {
  list: Interface_ListItem[];
}
interface Interface_ListItem {
  id: string;
}
interface Interface_Item {
  item: Interface_ListItem
}

// This one is simplified
type simplified = SimplifyDeep<Interface_Item> 

// This one contains a reference to Interface_ListItem
type not_simplified = SimplifyDeep<Interface_List>
```

type-fest@4.19.0

[Playground link](https://www.typescriptlang.org/play/?#code/PTAEGEHsAcEsFMAmoBmAnSBbUAXAntPALQrwDOOAAgCwB0AjAJy0AMAUPAB7SRo64F4ESADtEsHLFEBDADYBlWJmizYKPABF48aAB4AKoIA0oAKKcAxrICuieIcKgAvKBHwAbvDQmAkiKu29oLOoNYiANYikADuIgB8IQ5CXDjwYmRmljZ2SWwAkAD8oLl5AFzFwSlpiBl+ATmC+YWgAN4A2kkA0vB4oLAioOE9kCgVhAC65VBiElIicorKqupaOgaCHYLdeOMm5vVBhL7+2YfwcQC+TeVJANwc3Lz8+I6LKmqa2npJCS7T4pIZAolO8Vl91kdQAAxMIWQEDAA+oB8qTQ0gARrJ4LowpEYvETJB0QAreBwuL3NhsfqolDSCxCPy0+nwAD6ABlYBRWmxQKBVBRykyvHSGRyuTgUfBMG1xvcrjSRSzkSJmWLORQpdgWry+ohyhQ0P0AOby6mqpUMlVqtlanl8iTSoUWtCitkayWpTBsK5sEDFAAWXNAoiEwbIIOWCEQbBeQgjSzU0ZCb2WnzWwtdLNZWoSur5oH9AD0ClT-fogxlQ6ALKIcNJ+hlpKA0PBSK3-EIcJBrZb3RKtbHglEcKyE6Dky5Ux9VnpM27xRQ4vm+cXS0A)

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/sindresorhus) so you can upvote and help fund this issue.
- The funding will be given to active contributors.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/sindresorhus/type-fest/issues/884">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/sindresorhus/type-fest/issues/884/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/sindresorhus/type-fest/issues/884/pledge.svg">
</picture>
</a>

**Comment by KubaJastrz** [comment]
Apparently `Simplify` (the shallow one) also doesn't transform lists: https://github.com/sindresorhus/type-fest/issues/172#issuecomment-2120129618

**Comment by Emiyaaaaa** [comment]
fixed in #888

**Comment by sindresorhus** [comment]
Thank you @Emiyaaaaa for contributing to close this issue! ⭐

The rewards from this issue, totaling $10, has been shared with you.

**What now?**

1. Create a [Polar](https://polar.sh) account
2. See incoming rewards & setup Stripe to receive them
3. Get payouts as backers finalize their payments

_If you already have a Polar account setup, you don't need to do anything._
